plugins/destination/clickhouse/client/client_test.go

```diff
@@ -52,5 +53,17 @@ func TestPlugin(t *testing.T) {
 
 			MigrateStrategyAppend: migrateStrategy,
 		},
+		destination.WithTestSourceAllowNull(func(dt arrow.DataType) bool {
```

Comment:
Can you explain what's the reason we disallow null for these types but allow for others?

Reply:
Yes, it's this: https://clickhouse.com/docs/en/sql-reference/data-types/nullable
Array, Map & Tuple can't be nullable (arow: list, map, struct)